crates/turbopack-ecmascript/src/references/esm/export.rs

```diff
@@ -211,13 +235,44 @@ impl CodeGenerateable for EsmExports {
             match program {
                 Program::Module(Module { body, .. }) => {
                     body.insert(0, ModuleItem::Stmt(stmt));
+                    if let Some(cjs_stmt) = cjs_stmt.clone() {
+                        let loc = cjs_insertion_index(body);
+                        body.insert(loc, ModuleItem::Stmt(cjs_stmt));
+                    }
                 }
                 Program::Script(Script { body, .. }) => {
                     body.insert(0, stmt);
+
+                    if let Some(cjs_stmt) = cjs_stmt.clone() {
+                        let loc = cjs_insertion_index(body);
+                        body.insert(loc, cjs_stmt);
+                    }
                 }
             }
         }));
 
         Ok(CodeGeneration { visitors }.into())
     }
 }
+
+/// Insert the CJS export statement before the hoister directive. Practically,
+/// this places the cjs export statement right after the import statements.
+fn cjs_insertion_index<N>(stmts: &[N]) -> usize
+where
+    N: StmtLike,
+{
+    stmts
+        .iter()
+        .position(|s| {
+            if let Some(Stmt::Expr(ExprStmt { expr, .. })) = s.as_stmt() {
+                if let Expr::Lit(Lit::Str(Str { value, .. })) = &**expr {
+                    value == "__TURBOPACK__ecmascript__hoisting__location__"
```

Comment:
This magic value is `ESM_HOISTING_LOCATION` from `crates/turbopack-ecmascript/src/references/esm/base.rs`. Can we reference that directly, and possibly merge some code with `insert_hoisted_stmt`?